views/_includes/script.njk

```diff
@@ -26,4 +26,47 @@
 <script>
     if (typeof Promise !== "function" && document.querySelector('details') !== null)
         document.write('<script src="/js/details-element-polyfill.js"><\/script>');
+</script>
+
+<!-- Feedback form -->
+<script>
+    const form = document.getElementById('feedback-form');
+    const confirmation = document.getElementById('feedback-confirmation');
+    const error = document.getElementById('feedback-error');
+    const csrf = document.getElementById('feedback-csrf');
+
+    const submitFeedback = (event) => {
+        event.preventDefault();
+
+        let xmlhttp= window.XMLHttpRequest ?
+            new XMLHttpRequest() : new ActiveXObject("Microsoft.XMLHTTP");
+
+        // handle post-submit (swap form for confirmation)
+        xmlhttp.onreadystatechange = function() {
+            if (this.readyState === 4) {
+                form.classList.add('hidden');
+                if (this.status === 200) {
+                    confirmation.classList.remove('hidden');
+                } else {
+                    error.classList.remove('hidden');
+                }
+            }
+        }
+
+        // gather up the checkbox values
+        var problems = []
+        var checkboxes = form.querySelectorAll('input[type=checkbox]:checked')
+
+        for (var i = 0; i < checkboxes.length; i++) {
+            problems.push(checkboxes[i].value)
+        }
+
+        // post to our feedback endpoint
+        xmlhttp.open("POST","/en/feedback",true);
+        xmlhttp.setRequestHeader("Content-type","application/x-www-form-urlencoded");
+        xmlhttp.send("problems=" + problems + "&_csrf=" + csrf.value);
+    }
```

Comment:
Why are we doing this client side? Can't we add a new form element on each page and just post to a feedback controller?

Reply:
If javascript isn't available, that's what it'll do ... I had gone back and forth on this and opted to make it work the way it does on Canada.ca. The only issue I ran into trying to get it going with a straight postback was that the flash messaging system in node starter is not super flexible, so getting a confirmation message was a pain, so I bailed on it.

Reply:
Just going to add a feedback-thanks screen for non-javascript submission

Reply:
Sounds good.!